### PR TITLE
Chore: Refactor internals

### LINKED ISSUE #24: Adding multiple toasts in the same event loop breaks layout
If you run this code:

```ts
toast('hello')
toast('hello')
toast('hello')
```

then the layout will be messed up when hovering:

image

Creating them in a different event loop works:

```ts
toast('hello')
await new Promise((resolve) => resolve(null))
toast('hello')
await new Promise((resolve) => resolve(null))
toast('hello')
```

### PR DESCRIPTION
**Pull request by huntabyte**
These changes close #24 and from what I can tell #13 (not able to reproduce anymore with the provided example) and fixes a couple memory leaks with the prior implementation.

This PR refactors this project to leverage existing Svelte capabilities like stores, rather than implementing a custom observer.  By doing so, the internals have been greatly simplified. We no longer need to rely on dispatches between the `<Toast />` and `<Toaster />` to keep state in sync, as this can be done by subscribing to the `toasts` and `heights` stores within each of the components.

The original `sonner` added [props](https://sonner.emilkowal.ski/styling#tailwind-css) to add classes to the individual elements within the toast, which are passed via the `ToastOptions`, so those have been added to this project as well. 

[Vitest](https://vitest.dev) has been setup for unit testing the components. Only a couple tests have been added, but the pieces are in place to add more comprehensive tests that don't take 10 years to run (looking at you Playwright).

There shouldn't be any breaking changes to the public facing APIs, as the same methods and options are exposed, but we've added some additional functionality in the form of the (optional) additional `classes` property of `ToastOptions`.

[changesets](https://github.com/changesets/changesets) has been added along with a CI and release pipeline to make release notes and releases in general more seamless. Feel free to reach out on Discord if you'd like me to give you a more detailed rundown of how these work, but here's a quick guide to getting it setup:

1. Give appropriate permissions to GitHub actions within this projects settings
CleanShot 2023-12-28 at 23 37 28@2x
2. Add an NPM token with access to publish `svelte-sonner` to your repo's secrets called `NPM_TOKEN`
3. Add the [changesets bot](https://github.com/apps/changeset-bot) to your GitHub account
4. Enjoy :)

### CODE REVIEW

File: .changeset/config.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "$schema": "https://unpkg.com/@changesets/config@3.0.0/schema.json",
+  "changelog": "@changesets/cli/changelog",
+  "commit": false,
+  "fixed": [],
+  "linked": [],
+  "access": "restricted",
```
Comment (AdrianGonz97) [inline .changeset/config.json:7]:
```suggestion
  "access": "public",
```

File: package.json [hunk 2]
```diff
@@ -14,10 +14,12 @@
 		"package": "svelte-kit sync && svelte-package && publint",
 		"prepublishOnly": "npm run build",
 		"test": "playwright test",
+		"test:unit": "vitest",
 		"check": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json",
 		"check:watch": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json --watch",
 		"release": "bumpp && npm publish",
```
Comment (AdrianGonz97) [inline package.json:20]:
```suggestion
		"release": "pnpm run package && changeset publish",
```

File: package.json [hunk 3]
```diff
@@ -35,25 +37,39 @@
...
 		"publint": "^0.2.2",
 		"sass": "^1.67.0",
 		"svelte": "^4.2.0",
 		"svelte-check": "^3.5.2",
 		"svelte-preprocess": "^5.0.4",
 		"tslib": "^2.6.2",
 		"typescript": "^5.2.2",
-		"vite": "^4.4.9"
+		"vite": "^4.4.9",
+		"vitest": "^0.33.0"
```
Comment (AdrianGonz97) [inline package.json:72]: should probably just install the stable version
```suggestion
		"vitest": "^1.1.0"
```

**Review by AdrianGonz97 — CHANGES_REQUESTED** [review]

**Comment by wobsoriano** [comment]
Thank you for this! I plan to review it over the weekend. These improvements look fantastic.

I ported this project straight from the React component as a part of my Svelte learning journey. Excited to learn even more through your PR!

File: package.json (hunk 3 above)
Comment (wobsoriano) [reply package.json:72]: Let's go with latest version

**Review by wobsoriano — APPROVED** [review]
Love the use of svelte stores instead of making our own observers. Everything looks good so far. I have no major requests so I'm ready to merge.

**Comment by wobsoriano** [comment]
I had to manually release this one as I forgot to reinstall deps (since we updated vite version).

https://github.com/wobsoriano/svelte-sonner/actions/runs/7358434671/job/20031785729

**Comment by robinbraemer** [comment]
@huntabyte do you plan to add sonner toast to shadcn/ui-svelte? I’m hoping to see the new shadcn components ported to svelte. Awesome work everyone!

**Comment by AyushSehrawat** [comment]
> @huntabyte do you plan to add sonner toast to shadcn/ui-svelte? I’m hoping to see the new shadcn components ported to svelte. Awesome work everyone!

Yup it's coming to shadcn soon!

**Pull request merged**